.github/workflows/end-to-end.yml

```diff
@@ -0,0 +1,110 @@
+name: End-to-end Tests
+on:
+  pull_request:
+    paths:
+      - '*.go'
+      - '.revive.toml'
+      - 'cmd/**'
+      - 'go.mod'
+      - 'go.sum'
+      - 'Makefile'
+      - 'pkg/**'
+      - 'tools/**'
+      - 'config/**'
+      - '!config/stack/ttn-lw-stack.yml'
+      - 'package.json'
+      - 'pkg/webui/**'
+      - 'sdk/js/**'
+      - 'yarn.lock'
+
+jobs:
+  end-to-end:
+    name: Frontend based (cypress)
+    runs-on: ubuntu-18.04
+    env:
+      should_cache: true # Might be useful for debugging.
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v1
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(npx yarn cache dir)"
+        if: env.should_cache
+      - name: Apply yarn cache
+        uses: actions/cache@v2
+        id: yarn-cache
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-cache-${{ hashFiles('**/yarn.lock') }}-${{ hashFiles('**/package.json') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-cache-
+        if: env.should_cache
+      - name: Apply npm cache
+        uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+        if: env.should_cache
+      - name: Apply go modules cache
+        uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+        if: env.should_cache
+      - name: Apply database dump cache
+        uses: actions/cache@v2
+        with:
+          path: .cache/sqldump.sql
+          key: ${{ runner.os }}-sql-dump-cache-${{ hashFiles('pkg/identityserver/store/**/*.go') }}
+        if: env.should_cache
+      - name: Apply messages and locales cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            .cache/babel
+            .cache/messages
+            pkg/webui/console/locales/.backend
```

Comment:
This seems useful for other development builds as well, please copy-paste in all other workflows where we might need this